### PR TITLE
[community] Improve NFT flow UX.

### PR DESCRIPTION
**Pull request by zacharydenton**
### Description

- Always show the Connect Wallet button on load
- Inform the user that an Aptos wallet needs to be installed, and link to Petra
- Inform the user that the right network needs to be selected, and show an error if it is not
- Inform the user that the account needs to be funded, and show an error if it is not
- Open "Share on Twitter" in a new tab
- Show the truncated sender address in the NFT details

### Test Plan
Manual.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3962)

### CODE REVIEW

**Review by moonclavedev — APPROVED** [review]

File: ecosystem/platform/server/app/views/nft_offers/show.html.erb [hunk 1]
```diff
@@ -44,7 +44,7 @@
                 <% if step.name == :sign_in %>
                   To begin, sign in to Aptos.
                 <% elsif step.name == :connect_wallet %>
-                  Connect your wallet.
+                  Connect your wallet. You will need an <%= render(LinkComponent.new(href: 'https://chrome.google.com/webstore/detail/petra-aptos-wallet/ejjladinnckdgjemekebdpeokbikhfci')) { 'Aptos Wallet' } %> with a funded account and the network set to <%= @nft_offer.network.capitalize %>.
```
Comment (kent-white) [inline ecosystem/platform/server/app/views/nft_offers/show.html.erb:47]: nit: `Petra Wallet`

**Review by kent-white — APPROVED** [review]

File: ecosystem/platform/server/app/views/nft_offers/show.html.erb (hunk 1 above)
Comment (kent-white) [reply ecosystem/platform/server/app/views/nft_offers/show.html.erb:47]: Or maybe you are preparing for other wallets to integrate

Comment (zacharydenton) [reply ecosystem/platform/server/app/views/nft_offers/show.html.erb:47]: Yeah I want to keep it generic. Joe mentioned that we could have a modal that pops up with a grid of all the supported wallets.

Comment (zacharydenton) [reply ecosystem/platform/server/app/views/nft_offers/show.html.erb]: Updated copy to keep it generic but mention Petra specifically, per Max's suggestion.

**Pull request merged**